compiler+runtime/src/cpp/jank/runtime/obj/cons.cpp

```diff
@@ -77,12 +77,12 @@ namespace jank::runtime::obj
         for(auto it(fresh_seq()); it != nullptr;
             it = runtime::next_in_place(it), seq = runtime::next_in_place(seq))
         {
-          if(seq == nullptr || !runtime::equal(it, seq->first()))
+          if(seq == nullptr || !runtime::equal(it->first(), seq->first()))
```

Comment:
This seems like a good fix. We'd need the same thing for `chunked_cons` above, though. I wonder where else this cropped up. I have a TODO for sequence equality having a lot of copy pasta.

Reply:
> We'd need the same thing for chunked_cons above, though.

Done.

> I wonder where else this cropped up.

Several places. I think I reverted some of them accidentally when experimenting with replace `nullptr` with `nil_const()`, I'll have another look.

> I have a TODO for sequence equality having a lot of copy pasta.

I wasn't sure how to tackle it, but I can at least write more tests so then the abstraction can be validated easily.

Reply:
Don't worry about removing support for `nullptr` yet. Certainly not in this PR, which already has a lot going on.

Reply:
I have some ideas for how to do it. The tests are very welcome, though.